### PR TITLE
[SYCL][ESIMD] Add tests on simd load from accessors

### PR DESCRIPTION
**Pull request by yuriykoch**
Based on idea that offset should be a multiplier of alignment flag, and
alignment flag should be less strict than simd element alignment itself
for simd load constructor to work as expected.

Signed-off-by: Kochetkov, Yuriy <yuriyx.kochetkov@intel.com>

### CODE REVIEW

**Comment by yuriykoch** [comment]
Rebased over the latest `intel` branch to have all changes from #917

@vasilytric Would you mind to take a look? I've tried to split changes in `ctor_load_usm` tests on per-commit basis to make review easier.

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (vasilytric) [inline SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:165]: Do we need the `run` function?

Comment (yuriykoch) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:165]: The intent here is to have a single place where the mapping between `operator()` parameters and `LoadCtorAccTestDescription` parameters is done. Anything after `/* The rest are the test description parameters */` is passed as-is into the test  case description within the `run()`, so we don't need to manually handle parameter order each time we construct the test case description:
```
log::trace<TestDescriptionT>(std::forward<TestDescriptionArgsT>(args)...);
```

BTW, looks like I've left template parameters for `operator()` unintentionally. It's definitely confusing, sorry for that - will fix ASAP.

Comment (yuriykoch) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:165]: Removed template parameters for the `operator()` in 46f0ea82b55123bb5abd90243d280057bea8c51f

Comment (vasilytric) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:165]: ok

**Comment by yuriykoch** [comment]
@sndmitriev Would you mind to take a look?

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (yuriykoch) [inline SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:33]: There should be a dependency on `alignof(DataT)` also due to such restriction for simd load constructor

**Comment by yuriykoch** [comment]
As there is a restriction on possible `offset` values based on `alignof(DataT)` I would need to fix code a bit after merging #926 
Converted to the draft till then.

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (yuriykoch) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:33]: Done by 0f0c4edc59fb9992af48341442da22487c923987

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (vasilytric) [inline SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:135]: This comment is required?

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (vasilytric) [inline SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:140]:
```suggestion
  static constexpr int NumElems= NumElemsT::value;
```
What do you think about this?
`NumElems` used in all other tests

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (vasilytric) [inline SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:194]:
```suggestion
           "Reference data size is not equal to the simd vector size.");
```

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (vasilytric) [inline SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp]:
```suggestion
          // This alignment affect only the internal simd storage
```

Comment (yuriykoch) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp]: Thanks, will fix ASAP

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (yuriykoch) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:194]: I was using the term from the [documentation](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_intel_esimd/sycl_ext_intel_esimd.md#simd-vector-class) here

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (yuriykoch) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:140]: Fair, will fix ASAP

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (yuriykoch) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:135]: I was thinking about giving a hint about what `run_test` is doing and where the main logic is placed

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (vasilytric) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:140]: Or you could leave current variant to be more specific that this variable related to `simd` size

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (vasilytric) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:194]: ok, no problems to leave current variant

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp
Comment (vasilytric) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_acc.hpp:135]: Ok, no problem

**Comment by yuriykoch** [comment]
@kbobrovs Would you mind to take a look?

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc_core.cpp
Comment (kbobrovs) [inline SYCL/ESIMD/api/functional/ctors/ctor_load_acc_core.cpp:54]: I think oword should be one of the alignment cases as well. Is overal guaranteed to be a multiple of oword?

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc_core.cpp
Comment (kbobrovs) [inline SYCL/ESIMD/api/functional/ctors/ctor_load_acc_core.cpp:69]: This is really cool - the robustness of the infra to create new test combinations!

**Review by kbobrovs — APPROVED** [review]

**Comment by kbobrovs** [comment]
@vladimirlaz, the failure is unrelated - this is a clang++ crash on test compilation. Recently internal tickets were submitted on similar cases  - e.g. by @dongkyunahn-intel. Please merge.

Failed Tests (1):
  SYCL :: ESIMD/fp_args_size/fp_args_char_int_size_512.cpp

File: SYCL/ESIMD/api/functional/ctors/ctor_load_acc_core.cpp
Comment (yuriykoch) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_acc_core.cpp:54]: It's equal to `oword`. Though maybe it's better to remove default parameter at all and make alignment used explicit.

It would be a tiny fix just moving code around, so I'll make it in a separate PR today to ensure you could take a look at the suggestion without blocking this PR's merge. Thanks for the catch.

**Pull request merged**
